lib/blocks.php

```diff
@@ -504,3 +504,94 @@ function gutenberg_migrate_old_typography_shape( $metadata ) {
 if ( ! function_exists( 'wp_migrate_old_typography_shape' ) ) {
 	add_filter( 'block_type_metadata', 'gutenberg_migrate_old_typography_shape' );
 }
+
+/**
+ * Enqueue a stylesheet for a specific block.
+ *
+ * If the theme has opted-in to separate-styles loading,
+ * then the stylesheet will be enqueued on-render,
+ * otherwise when the block inits.
+ *
+ * @param string $block_name The block-name, including namespace.
+ * @param array  $args       An array of arguments [handle,src,deps,ver,media].
+ * @param bool   $register   Whether the stylesheet should be registered.
+ *                           If set to false, then assume it has already been registered,
+ *                           and only enqueue it.
+ *
+ * @return void
+ */
+function gutenberg_enqueue_block_style( $block_name, $args, $register = true ) {
+	$args = wp_parse_args(
+		$args,
+		array(
+			'handle' => '',
+			'src'    => '',
+			'deps'   => array(),
+			'ver'    => false,
+			'media'  => 'all',
+		)
+	);
+
+	/**
+	 * Callback function to register and enqueue styles.
+	 *
+	 * @param string $content When the callback is used for the render_block filter,
+	 *                        the content needs to be returned so the function parameter
+	 *                        is to ensure the content exists.
+	 *
+	 * @return string
+	 */
+	$callback = function( $content ) use ( $args, $register ) {
+		// Register the stylesheet.
+		if ( $register ) {
```

Comment:
Can we detect whether the style needs to be registered based on the `$args` passed? My assumption would be that if there is nor `src`, `deps`, `ver` or `media` passed then we can assume it's a handle only. Alternatively, we could pass the handle name as `$args`. The third `$register` boolean param seems redundant in my opinion.

Reply:
Good point. Fixed :+1: